src/main/java/com/aws/greengrass/mqttbridge/MQTTBridge.java

```diff
@@ -87,35 +86,52 @@ protected MQTTBridge(Topics topics, TopicMapping topicMapping, MessageBridge mes
         this.pubSubClient = new PubSubClient(pubSubIPCAgent);
         this.ioTCoreClient = new IoTCoreClient(iotMqttClient);
         this.executorService = executorService;
-    }
 
-    @Override
-    public void install() {
-        mappingConfigTopics =
-                this.config.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY, MQTT_TOPIC_MAPPING);
-
-        mappingConfigTopics.subscribe((whatHappened, node) -> {
-            if (mappingConfigTopics.isEmpty()) {
-                logger.debug("Mapping empty");
-                topicMapping.updateMapping(Collections.emptyMap());
+        // handle configuration changes
+        Topics mappingConfigTopics = topics.lookupTopics(BridgeConfig.PATH_MQTT_TOPIC_MAPPING);
+        topics.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY).subscribe((what, child) -> {
+            // initialization
```

Comment:
in the subscribe, ignore the unimportant changes like GenericExternalService does that I sent you. timestamp for example should be ignored

Reply:
updated 👍